assets/src/dashboard/app/views/templates/detail.js

```diff
@@ -186,39 +206,54 @@ function TemplateDetail() {
     template && (
       <FontProvider>
         <TransformProvider>
-          <TemplateNavBar />
-          <ContentContainer>
-            <ColumnContainer>
-              <Column>
-                {PrevButton}
-                <CardGallery>{previewPages}</CardGallery>
-              </Column>
-              <Column>
-                <DetailContainer>
-                  <Title>{template.title}</Title>
-                  <ByLine>{byLine}</ByLine>
-                  <Text>{template.description}</Text>
-                  <MetadataContainer>
-                    {template.tags.map((tag) => (
-                      <Pill
-                        name={tag}
-                        key={tag}
-                        disabled
-                        onClick={() => {}}
-                        value={tag}
-                      >
-                        {tag}
-                      </Pill>
-                    ))}
-                  </MetadataContainer>
-                  <MetadataContainer>
-                    <ColorList colors={template.colors} size={30} />
-                  </MetadataContainer>
-                </DetailContainer>
-                {NextButton}
-              </Column>
-            </ColumnContainer>
-          </ContentContainer>
+          <UnitsProvider pageSize={pageSize}>
```

Comment:
Nit, `UnitsProvider` isn't needed this high up the tree. `CardGallery` actually uses its own `UnitsProvider` to calculate the dimensions for the story cards in the gallery.

Since the "Related Templates" section is gated behind a flag (so it'll appear or not appear) and only related templates is what needs the `UnitsProvider`, you could wrap the `RowContainer` or even just the `StoryGridView` with `UnitsProvider` to keep only what needs it getting that provider action 😎